### PR TITLE
string-util: check for overflow in strrep()

### PR DESCRIPTION
**Pull request by mrc0mmand**
This simply mirrors the same overflow check we already have in strrepa(), in case someone passed us a sufficiently long string.

strrep() is currently used only in tests, so this is just hardening.

### CODE REVIEW

**Review by bluca — APPROVED** [review]

File: src/basic/string-util.c [hunk 1]
```diff
@@ -984,7 +984,10 @@ char* strrep(const char *s, unsigned n) {
         assert(s);
 
         l = strlen(s);
-        p = r = malloc(l * n + 1);
+        if (!MUL_ASSIGN_SAFE(&l, n))
+                return NULL;
+
```
Comment (poettering) [inline src/basic/string-util.c:991]: hmm, I welcome such overflow checks, but let's maybe also cover the addition below too?

i.e. use INC_SAFE to add the 1 extra value.

Comment (poettering) [reply src/basic/string-util.c:991]: hmm, and if you touch this, you'd be very welcome two make two modernization changes:

1. change the type of `n` and `i` to `size_t` we generally want that values that count memory objects should be size_t, and this does.
2. please rename `r` to something else, maybe `ret` or `b` or something. We generally use 'int r;` everywhere for the error return variable, all across our tree. And suddenly having a function here that names something else `r` is a bit weird...

but neither of these two issues are really relevant, but still very welcome changes.

Comment (mrc0mmand) [reply src/basic/string-util.c:991]: Thanks, this indeed makes sense, and it also makes adding a test case for the overflow case much easier.

**Comment by poettering** [comment]
perfect! thanks!
